contrib/ansible/vagrant/Vagrantfile

```diff
@@ -1,46 +1,100 @@
 # -*- mode: ruby -*-
 # vi: set ft=ruby :
+require 'date'
 
-# All Vagrant configuration is done below. The "2" in Vagrant.configure
-# configures the configuration version (we support older styles for
-# backwards compatibility). Please don't change it unless you know what
-# you're doing.
-Vagrant.configure(2) do |config|
-  config.vm.box = "chef/centos-7.0"
+### This is a new provider, different then cloudbau's.  
+### RUN: vagrant plugin uninstall vagrant-openstack-plugin"
+### Then RUN: "vagrant plugin install vagrant-openstack-provider"
+require 'vagrant-openstack-provider'
 
-#  config.vm.network "public_network"
+Total=3
 
-  config.vm.define "master", primary: true do |master|
-    master.vm.hostname = "master.vms.local"
-    master.vm.network "private_network", ip: "192.168.1.100"
+VAGRANTFILE_API_VERSION = "2"
+
+# Openstack + Hostmanager providers are best used with latest versions.
+Vagrant.require_version ">= 1.7"
+
+### If you want to change variables in all.yml, use this snippet.
+### Just add a new line below as necessary...
+text = File.read('../group_vars/all.yml')
+new_contents = text.gsub("#ansible_ssh_user: root", "ansible_ssh_user: fedora")
```

Comment:
it could be a solution. I think making it totally generic is pretty complex to do, I'd prefer to start off with a complete impl in one env for the sake of history and sanity, and evolve it over time, and open stack is nice since it doesn't make my laptop catch on fire when deploying 4 node clusters